packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -228,12 +202,27 @@ class ViewportPositioner extends Foundation<
     }
 
     public componentDidUpdate(prevProps: ViewportPositionerProps): void {
+        // if anchor or viewport changes the component needs to reset
+        if (
+            prevProps.anchor !== this.props.anchor ||
+            prevProps.viewport !== this.props.viewport
+        ) {
+            this.disconnectObservers();
```

Comment:
could you add a method for resetting the component and just run both of these together? seems like there's no case in which you aren't doing both `disconnectObservers` and `disconnectViewport`.

Reply:
Merged into "detachListeners"